Kickstarter-iOS/ViewModels/RootViewModel.swift

```diff
@@ -244,9 +250,62 @@ internal final class RootViewModel: RootViewModelType, RootViewModelInputs, Root
       .skipNil()
       .map { $0 as RootViewControllerIndex }
 
-    self.setBadgeValueAtIndex = activityViewControllerIndex
-      .takeWhen(self.viewDidLoadProperty.signal)
-      .map { index in (activitiesBadgeValue(), index) }
+    let lifecycleEvents = Signal.merge(
+      self.viewDidLoadProperty.signal,
+      self.applicationWillEnterForegroundSignal
+    )
+
+    let updateBadgeValueOnLifecycleEvents = activityViewControllerIndex
+      .takeWhen(lifecycleEvents)
+      .map { index in
+        (activitiesBadgeValue(with: AppEnvironment.current.application.applicationIconBadgeNumber), index)
+      }
+
+    let selectedIndexAndActivityViewControllerIndex = Signal.combineLatest(
+      self.selectedIndex,
+      activityViewControllerIndex
+    )
+
+    let clearBadgeValueOnActivitiesTabSelected = selectedIndexAndActivityViewControllerIndex
+      .filter(==)
+      .on(
+        value: { _ in
+          _ = AppEnvironment.current.apiService.clearUserUnseenActivity(input: EmptyInput())
```

Comment:
Open to suggestions on how to test this. Currently it's just injected as a side-effect when the tab bar item is selected. I don't really want to create a void output simply to say 'this emitted' as it causes no update to the UI if it succeeds or fails. One option I thought of is to track it through Koala and assert that the event was tracked?

Reply:
After I clear the count, I update the user and test that it emits again. Would that work here?

Reply:
Ah, yes I think that makes sense! Thanks for the suggestion. For the most part I was just relying on mostly having the badge value from push but actually I should refresh the user to avoid an inconsistent state.

Reply:
Can the mutation return the new `unseenActivityCount`? If so can we just set it on the current user? 🤷‍♀

Reply:
Yeah I'm down with using the response from the mutation 👍

Reply:
Ok, this now uses the value returned from the mutation and updates the current user on `AppEnvironment`. This is a "fire and forget" call so we don't flip the value back if the call fails.